File: Python/subprogramas/convertir_a_decimal.py

def convertir_a_decimal(binario):
    decimal = 0
    for i in range(len(binario)):
        decimal += int(binario[i]) * (2 ** (len(binario) - i - 1))
    return decimal

# Example usage
if __name__ == "__main__":
    binario = "1010"
    print(f"El binario {binario} en decimal es {convertir_a_decimal(binario)}")

File: C++/subprogramas/convertir_a_decimal.cpp

#include <iostream>
#include <string>

int convertir_a_decimal(const std::string& binario) {
    int decimal = 0;
    for (size_t i = 0; i < binario.size(); ++i) {
        decimal += (binario[i] - '0') * (1 << (binario.size() - i - 1));
    }
    return decimal;
}

int main() {
    std::string binario = "1010";
    std::cout << "El binario " << binario << " en decimal es " << convertir_a_decimal(binario) << std::endl;
    return 0;
}